lib/src/lints/avoid_debug_print_in_release/models/avoid_debug_print_in_release_debug_model.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+
+/// A class used to parse function expression
+class AvoidDebugPrintInReleaseCheckModel {
+  /// Wheather the statement is prefixed with `!`
+  final bool negated;
+
+  /// Function name
+  final String name;
+
+  /// Function's source path
+  final String sourcePath;
+
+  /// A class used to parse function expression
+  const AvoidDebugPrintInReleaseCheckModel({
+    required this.negated,
+    required this.name,
+    required this.sourcePath,
+  });
+
+  /// A constructor that parses identifier into [name] and [sourcePath]
+  factory AvoidDebugPrintInReleaseCheckModel.parseExpression(
+    Expression? expression,
+  ) {
+    switch (expression?.childEntities.toList()) {
+      case [
+          final Token token,
+          final Identifier identifier,
+        ]:
+        return AvoidDebugPrintInReleaseCheckModel._parseIdentifier(
+          identifier: identifier,
+          negated: token.type == TokenType.BANG,
+        );
+      case [final Identifier identifier]:
+        return AvoidDebugPrintInReleaseCheckModel._parseIdentifier(
+          identifier: identifier,
+          negated: false,
+        );
+      default:
+        return AvoidDebugPrintInReleaseCheckModel._empty();
+    }
+  }
+
+  factory AvoidDebugPrintInReleaseCheckModel._empty() {
+    return const AvoidDebugPrintInReleaseCheckModel(
+      negated: false,
+      name: '',
+      sourcePath: '',
+    );
+  }
+
+  static const String _expectedPath =
+      'package:flutter/src/foundation/constants.dart';
+
+  static const String _kReleaseMode = 'kReleaseMode';
+
+  /// Whether the expression has the same source library as kDebugMode
+  bool get hasTheSameSource => _expectedPath == sourcePath;
+
+  /// Whether the expression has the same name as kReleaseMode
+  bool get hasSameName => _kReleaseMode == name;
```

Comment:
```suggestion
  bool get _isKReleaseModeName => _kReleaseMode == name;
```
I think this name would be more clear.